File: web/HappyMoney/HappyMoney/ViewModels/EditBudgetViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using HappyMoney.Models;

namespace HappyMoney.ViewModels
{
	public class EditBudgetViewModel
	{
		public EditBudgetViewModel(Budget budget)
		{
			if (budget == null)
			{
				throw new ArgumentNullException("budget");
			}

			this.Name = budget.Name;
			this.Capacity = budget.Envelopes.Sum(env => env.Capacity);
			this.Guid = budget.Guid;
		}

		public string Name { get; private set; }
		public double Capacity { get; private set; }
		public Guid Guid { get; private set; }
	}
}

File: web/HappyMoney/HappyMoney/Scripts/viewModels/EditBudgetViewModel.js
define(['knockout', 'models/envelope'], function (ko, envelopeService) {
    var EnvelopeViewModel = function (env) {
        this.name = ko.observable(env.name);
        this.capacity = ko.observable(env.capacity);
        this.guid = env.guid;

        var self = this;
        this.toPojo = function () {
            return {
                name: self.name(),
                capacity: self.capacity(),
                guid: self.guid
            };
        }
    }

    var EditBudgetViewModel = function (data) {
        var self = this;

        this.envelopes = ko.observableArray(data.map(function (env) {
            return ko.observable(new EnvelopeViewModel(env));
        }));

        this.capacity = ko.computed(function () {
            return self.envelopes().reduce(function (accum, curr) {
                return accum + parseFloat(curr().capacity());
            }, 0.0);
        });

        this.saveChanges = function () {
            envelopeService.updateEnvelopes(this.envelopes().map(function (env) { return env().toPojo(); }));
        };
    };

    return EditBudgetViewModel;
});